lib/Config.js

```diff
@@ -243,6 +243,23 @@ class Config {
             }
         }
 
+        this.metadataDaemon = {};
+        if (config.metadataDaemon) {
+            if (config.metadataDaemon.host !== undefined) {
```

Comment:
Either you would create the defaults or ensure that both host and port are defined here. The if conditionals allow one of them to be defined and the other undefined.